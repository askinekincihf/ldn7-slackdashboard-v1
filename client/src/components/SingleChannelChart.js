import { Bar } from "react-chartjs-2";

const SingleChannelChart = (props) => {
	const state = {
		data: {
			labels: props.label,
			datasets: [
				{
					label: "Messages",
					backgroundColor: "#ffa600",
					borderColor: "#003f5c",
					borderWidth: 1,
					hoverBackgroundColor: "#ff7c43",
					hoverBorderColor: "#003f5c",
					data: props.messagesDataSet,
				},

				{
					label: "Reactions",
					backgroundColor: "#ff6361",
					borderColor: "#003f5c",
					borderWidth: 1,
					hoverBackgroundColor: "#d45087",
					hoverBorderColor: "#003f5c",
					data: props.reactionsDataSet,
				},
			],
		},
	};
	const options = {
		responsive: true,
		legend: {
			display: false,
		},
		type: "bar",
	};
	return (
		<div className="chart">
			<Bar data={state.data} options={options} />
		</div>
	);
};

export default SingleChannelChart;
