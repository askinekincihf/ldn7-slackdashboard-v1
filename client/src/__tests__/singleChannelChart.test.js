import React from "react";
import renderer from "react-test-renderer";
import SingleChannelChart from "../components/SingleChannelChart";

describe("SingleUserChart", () => {
	it("should render a chart for a single channel correctly", () => {
		const component = renderer.create(<SingleChannelChart />).toJSON();

		expect(component).toMatchSnapshot();
	});
});
