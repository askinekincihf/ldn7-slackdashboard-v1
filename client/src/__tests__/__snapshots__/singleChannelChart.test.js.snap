// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleUserChart should render a chart for a single channel correctly 1`] = `
<div
  className="singleChannelChart"
>
  <canvas
    data-testid="canvas"
    height={150}
    onClick={[Function]}
    role="img"
    width={300}
  />
</div>
`;
